packages/backend/src/modules/finality/analyzers/types/BaseAnalyzer.ts

```diff
@@ -88,5 +123,69 @@ export abstract class BaseAnalyzer {
    * @param transaction
    * @returns TTI/SUD delays in seconds for each transaction
    */
-  abstract analyze(transaction: Transaction): Promise<number[]>
+  abstract analyze(
+    previousTransaction: Transaction,
+    transaction: Transaction,
+  ): Promise<L2Block[]>
+}
```

Comment:
Maybe we can mae previousTransaction optional as we do not even use it in T2I analyzers

Reply:
I don't want to make it optional because the `BaseAnalyzer.ts` is always passing it down